src/server/builder.ts

```diff
@@ -0,0 +1,417 @@
+﻿/// <reference path="..\compiler\commandLineParser.ts" />
+/// <reference path="..\services\services.ts" />
+/// <reference path="protocol.d.ts" />
+/// <reference path="session.ts" />
+/// <reference types="node" />
+
+namespace ts.server {
+
+    interface Hash {
+        update(data: any, input_encoding?: string): Hash;
+        digest(encoding: string): any;
+    }
+
+    const crypto: {
+        createHash(algorithm: string): Hash
+    } = require("crypto");
+
+    /**
+     * An abstract file info that maintains a shape signature.
+     */
+    export class BuilderFileInfo {
+
+        private lastCheckedShapeSignature: string;
+
+        constructor(

```

Comment:
can you put arguments on one line? there are only two of them so it should still look ok. Currently the sequence 
`) {
}`
look a bit odd